### PR TITLE
Add schema migration test

### PR DESCRIPTION
**Pull request by Ishiihara**
@ewencp Added schema migration test.

### CODE REVIEW

**Comment by Ishiihara** [comment]
This depends on https://github.com/confluentinc/kafka-connect-hdfs/pull/21

File: muckrake/tests/connect_hdfs_test.py [hunk 1]
```diff
@@ -174,4 +174,80 @@ def check_files(self):
...
+
+        index = 0
+        for line in node.account.ssh_capture(cmd):
+            if line.strip() != '':
+                get_schema_cmd = "HADOOP_CONF_DIR=/mnt " + hadoop_home + "/bin/hadoop jar " + \
+                               "/opt/avro/" + avro_tools + " getschema " + line.strip(" |\n") + "| tail -n +2 "
+                json_string = node.account.ssh_output(get_schema_cmd)
+                json_data = json.loads(json_string.strip("|\n"))
+                self.logger.debug(json_data)
+                if index == 0:
```
Comment (ewencp) [inline muckrake/tests/connect_hdfs_test.py]: Can we document how this test actually works? I was confused by this, and now I want to make sure I understand correctly. Does this explanation make sense:

The source generates messages with alternating schema versions, i.e. 1, 2, 1, 2, ... In backwards compatibility mode, when we encounter a new schema we should immediately rotate to a new file and then project to the new schema from then on. Therefore, we should end up with 1 file which will have only 1 entry for the first messages with schema 1, then the next message will force us to switch to schema 2. From then on we should always be on schema 2 in subsequent files.

File: muckrake/tests/connect_hdfs_test.py [hunk 2]
```diff
@@ -174,4 +174,80 @@ def check_files(self):
...
+
+        index = 0
+        for line in node.account.ssh_capture(cmd):
+            if line.strip() != '':
+                get_schema_cmd = "HADOOP_CONF_DIR=/mnt " + hadoop_home + "/bin/hadoop jar " + \
+                               "/opt/avro/" + avro_tools + " getschema " + line.strip(" |\n") + "| tail -n +2 "
+                json_string = node.account.ssh_output(get_schema_cmd)
+                json_data = json.loads(json_string.strip("|\n"))
+                self.logger.debug(json_data)
+                assert index % 2 + 1 == int(json_data['connect.version'])
```
Comment (ewencp) [inline muckrake/tests/connect_hdfs_test.py]: This is also probably clear given the previous explanation, but could still use a note.

**Comment by ewencp** [comment]
@Ishiihara LGTM

**Pull request merged**